kliff/dataset/dataset.py

```diff
@@ -301,67 +499,207 @@ def order_by_species(self):
-        configurations.
-        This is a convenience function to read configurations from different directory
-        on disk.
+        Read configurations from colabfit database.
 
         Args:
-            path: Path the directory (or filename) storing the configurations.
+            database_client: Instance of connected MongoDatabase client, which can be used to
+                fetch database from colabfit-tools dataset.
+            colabfit_dataset: Name of the colabfit dataset instance to read from.
             weight: an instance that computes the weight of the configuration in the loss
                 function.
+
+        Returns:
+            A list of configurations.
         """
+        # get configuration and property ID and send it to load configuration-first get Data Objects
+        dataset_dos = database_client.get_data(
+            "data_objects",
+            fields=["colabfit-id"],
+            query={"relationships.datasets": {"$in": [colabfit_dataset]}},
+        )
+        if not dataset_dos:
+            logger.error(f"{colabfit_dataset} is either empty or does not exist")
+            raise DatasetError(f"{colabfit_dataset} is either empty or does not exist")
+
+        configs = []
+        for do in dataset_dos:
+            co_doc = database_client.configurations.find_one(
+                {"relationships.data_objects": {"$in": [do]}}
+            )
+            pi_doc = database_client.property_instances.find(
+                {"relationships.data_objects": {"$in": [do]}}
+            )
+            co_id = co_doc["colabfit-id"]
+            pi_ids = [i["colabfit-id"] for i in pi_doc]
 
-        configs = self._read(path, weight, self.file_format)
-        self.configs.extend(configs)
+            configs.append(
+                Configuration.from_colabfit(database_client, co_id, pi_ids, weight)
+            )
+            # TODO: reduce number of queries to database. Current: 4 per configuration
 
-    def get_configs(self) -> List[Configuration]:
+        if len(configs) <= 0:
+            raise DatasetError(f"No dataset file with in {colabfit_dataset} dataset.")
+
+        logger.info(f"{len(configs)} configurations read from {colabfit_dataset}")
+
+        return configs
+
+    def add_from_colabfit(
+        self,
+        colabfit_database: str = None,
+        colabfit_dataset: str = None,
+        weight: Optional[Weight] = None,
+    ):
         """
-        Get the configurations.
+        Read configurations from colabfit database and add them to the dataset.
+
+        Args:
+            colabfit_database: Name of the colabfit Mongo database to read from.
+            colabfit_dataset: Name of the colabfit dataset instance to read from.
+            weight: an instance that computes the weight of the configuration in the loss
+                function.
+
         """
-        return self.configs
+        if colabfit_database is not None:
```

Comment:
We should remove the two checks `colabfit_database` and `colabfit_dataset`? If they are called, both values must be provided.

Reply:
Removed the check, as colabfit_dataset and database now will not be none, and a positional arguments.